deploy/vagrant/modules/buttonmen/manifests/init.pp

```diff
@@ -129,3 +129,28 @@
       hour => "0";
   }
 }
+
+class buttonmen::clienttestsetup {
+
+  package {
+    "python": ensure => installed;
+    "python3": ensure => installed;
+  }
+
+  exec {
+
+    "pip_install":
+      command     => "/usr/bin/wget -qO - https://bootstrap.pypa.io/pip/2.7/get-pip.py | /usr/bin/python",
+      require => [Package["wget"], Package["python"]];
+    "pip3_install":
+      command     => "/usr/bin/wget -qO - https://bootstrap.pypa.io/pip/3.5/get-pip.py | /usr/bin/python3",
+      require => [Package["wget"], Package["python"]];
+
+    "pip_requirements_install":
+      command     => "/usr/bin/python -m pip install -r /buttonmen/tools/python/api_client/requirements.txt",
+      require => [Exec["pip_install"]];
+    "pip3_requirements_install":
+      command     => "/usr/bin/python3 -m pip install -r /buttonmen/tools/python/api_client/requirements.txt",
+      require => [Exec["pip3_install"]];
+  }
+}
```

Comment:
maybe I shouldn't have done this client test setup in puppet as it then installs these tools in more places then it needs to be. maybe I should have done these as a step in circleci since that's the only place the actually needs these tests to run? ¯\_(ツ)_/¯ im looking for others who have more experience with designing the current vagrant/puppet/circleci/docker set up to weigh in and confirm or correct my decisions here

Reply:
I agree with the decision to do this in puppet, if we're going to add more mandatory dependencies.  It *might* mean this stuff works out of box for dev and replay sites, certainly makes it more likely.